src/odemis/driver/tlaptmf.py

```diff
@@ -374,27 +377,27 @@ def _ReadMessage(self, timeout=None):
             for i in range(6):
                 char = self._serial.read() # empty if timeout
                 if not char:
-                    raise IOError("Controller timed out, after receiving '%s'" % msg)
+                    raise IOError("Controller timed out, after receiving '%s'" % msg.decode('latin1', 'backslashreplace'))
 
                 msg += char
         finally:
             self._serial.timeout = old_timeout
 
         mid = struct.unpack("<H", msg[0:2])[0]
-        if not (ord(msg[4]) & 0x80): # short message
-            logging.debug("Received: '%s'", ", ".join("%02X" % ord(c) for c in msg))
+        if not (ord(msg.decode('latin1', 'backslashreplace')[4]) & 0x80): # short message
```

Comment:
Use [4:5], instead of the `decode()`, or convert `msg` to a bytearray. (Also below, in other similar patterns).